### ISSUE TITLE
OpenBLAS multithreading crash - Running MnistMLPExample error.

### ISSUE THREAD

**Issue by green-creeper** [deeplearning4j/deeplearning4j#2353]
Hi,
I'm trying to run dl4j example and I faced with the error which I cannot resolve. 
I use JDK8_u112 and macOS 10.12.

Here is Stdout and hs_err_pid50428.log

https://gist.github.com/green-creeper/7cdac62c09dcd1f0405c058ed7a33d3e

any suggestions?

UPD: The same problem with TrainingStatsExample and SparkLSTMCharacterExample

**Comment by saudet** [comment]
Ah, looks like we might have an answer here: https://github.com/xianyi/OpenBLAS/issues/539#issuecomment-92445199
The binaries were built with NUM_THREADS=4, so try to `export OMP_NUM_THREADS=4` before running DL4J.

**Comment by AlexDBlack** [comment]
I've hit a very similar or possibly identical issue, as discussed on gitter.
Here's the log after setting OMP_NUM_THREADS=4 (also crashed without that, using 12 threads)

https://gist.github.com/AlexDBlack/99760f1e6d965e36ffbef7eca6192193

**Comment by saudet** [comment]
I am currently using this script file to build OpenBLAS for the release:
https://github.com/bytedeco/javacpp-presets/blob/master/openblas/cppbuild.sh
I've fixed it accordingly, but would like to confirm that it solves this issue.
@green-creeper @AlexDBlack Could you try to rebuild and run that on your machines?
It's basically:
```bash
git clone https://github.com/bytedeco/javacpp-presets
export OpenBLAS_HOME=$PWD/javacpp-presets/openblas/cppbuild/your-platform/
cd javacpp-presets/openblas
bash cppbuild.sh install
```
And then build libnd4j and ND4J as usual.

**Comment by AlexDBlack** [comment]
Quick test of this un Ubuntu 16.04 + Spark MnistMLPExample: 0.7.0 crashes with default settings

New code seems more stable, but I can (unsuprisingly) easily make it crash by setting sparkConf.setMaster("local[256]"); in the example.

I'm doing a full run with local[12] (on a 6 core + HT machine - so equivalent to local[*] here). No issues after a couple of epochs; I'll update this comment once it's finished (or not).

**Comment by AlexDBlack** [comment]
Possibly related: see my comment here - https://github.com/deeplearning4j/deeplearning4j/issues/2366

**Comment by AlexDBlack** [comment]
Fixed as of 0.7.1
